packages/eslint-plugin/src/rules/prefer-record.ts

```diff
@@ -0,0 +1,118 @@
+import { createRule } from '../util';
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+
+export default createRule({
+  name: 'prefer-record',
```

Comment:
Not a full review - just a quick flyby comment

might be better to call this `consistent-indexed-object-style` instead, because it can enforce either way?